docker-compose.yml

```diff
@@ -0,0 +1,131 @@
+
+#
+# Usage:
+# ------
+#
+# This docker compose file parameterizes build configurations using environment
+# variables from the host. The default values for the build configuration
+# variables are set in `.env`.
+#
+# Variables:
+#   ARCH={amd64, arm64v8, ...}
+#   COMPILER_TYPE={gnu, clang, intel, nvidia}
+#   COMPILER={gcc-5, gcc-6, gcc-7, gcc-8, gcc-9, gcc-10,
+#             clang-3.9, clang-4.0, clang-5.0, clang-6.0, clang-7, clang-8,
+#             clang-9, clang-10,
+#             icc-18, icc-19,
+#             nvcc-10, nvcc-11}
+#   REPO=lifflander1/checkpoint
+#   UBUNTU={18.04}
+#   ULIMIT_CORE=0
+#
+# DARMA/checkpoint Configuration Variables:
+#   CHECKPOINT_DOCS=0    # Enable doxygen build
+#   BUILD_TYPE=release   # CMake build type
+#   CODE_COVERAGE=0      # Enable generation of code coverage reports
+#
+# In order to run in the container, there are two stages. First, one must build
+# or pull the base container with the variables exported (or using the defaults
+# in `checkpoint/.env` that configure it:
+#
+# For a non-interactive build with gcc-8, since gnu is the default compiler
+# type, one may do the following:
+#
+#  $ COMPILER=gcc-8 docker-compose pull ubuntu-cpp
+#  $ COMPILER=gcc-8 docker-compose run ubuntu-cpp
+#
+
+# Need verision >= 3.5 for the features in use
+version: '3.5'
+
+# Named volumes must be predefined according the docker compose rules. Many
+# combinations have already been added, but if a needed configuration is missing
+# add it to this list. For example, for ARM64v8 on Ubuntu 20.04 with clang-9,
+# add `arm64v8-ubuntu-20.04-clang-9-cache`.
+volumes:
+  amd64-ubuntu-18.04-clang-3.9-cache:
```

Comment:
do we need as many volumes here?

Reply:
I do not know much about Docker. 
This file is extracted from `vt/docker-compose.yml`
Maybe @lifflander could comment which useful.

Reply:
Done